node/external/blockAPI/metaBlock.go

```diff
@@ -152,25 +152,74 @@ func (mbp *metaAPIBlockProcessor) convertMetaBlockBytesToAPIBlock(hash []byte, b
-			TotalSupply:                      epochStartEconomics.TotalSupply.String(),
-			TotalToDistribute:                epochStartEconomics.TotalToDistribute.String(),
-			TotalNewlyMinted:                 epochStartEconomics.TotalNewlyMinted.String(),
-			RewardsPerBlock:                  epochStartEconomics.RewardsPerBlock.String(),
-			RewardsForProtocolSustainability: epochStartEconomics.RewardsForProtocolSustainability.String(),
-			NodePrice:                        epochStartEconomics.NodePrice.String(),
-			PrevEpochStartRound:              epochStartEconomics.PrevEpochStartRound,
-			PrevEpochStartHash:               hex.EncodeToString(epochStartEconomics.PrevEpochStartHash),
-		}
+	return apiMetaBlock, nil
+}
+
+func addStartOfEpochInfoInBlock(metaBlock *block.MetaBlock, apiBlock *api.Block) {
+	if !metaBlock.IsStartOfEpochBlock() {
+		return
+	}
+
+	epochStartEconomics := metaBlock.EpochStart.Economics
+
+	apiBlock.EpochStartInfo = &api.EpochStartInfo{
+		TotalSupply:                      epochStartEconomics.TotalSupply.String(),
+		TotalToDistribute:                epochStartEconomics.TotalToDistribute.String(),
+		TotalNewlyMinted:                 epochStartEconomics.TotalNewlyMinted.String(),
+		RewardsPerBlock:                  epochStartEconomics.RewardsPerBlock.String(),
+		RewardsForProtocolSustainability: epochStartEconomics.RewardsForProtocolSustainability.String(),
+		NodePrice:                        epochStartEconomics.NodePrice.String(),
+		PrevEpochStartRound:              epochStartEconomics.PrevEpochStartRound,
+		PrevEpochStartHash:               hex.EncodeToString(epochStartEconomics.PrevEpochStartHash),
+	}
+
+	if len(metaBlock.EpochStart.LastFinalizedHeaders) == 0 {
+		return
+	}
+
+	epochStartShardsData := metaBlock.EpochStart.LastFinalizedHeaders
+	apiBlock.EpochStartShardsData = make([]*api.EpochStartShardData, 0, len(epochStartShardsData))
+	for _, epochStartShardData := range epochStartShardsData {
+		addEpochStartShardDataForMeta(epochStartShardData, apiBlock)
+	}
+}
+
+func addEpochStartShardDataForMeta(epochStartShardData block.EpochStartShardData, apiBlock *api.Block) {
+	shardData := &api.EpochStartShardData{
+		ShardID:               epochStartShardData.ShardID,
+		Epoch:                 epochStartShardData.Epoch,
+		Round:                 epochStartShardData.Round,
+		Nonce:                 epochStartShardData.Nonce,
+		HeaderHash:            hex.EncodeToString(epochStartShardData.HeaderHash),
+		RootHash:              hex.EncodeToString(epochStartShardData.RootHash),
+		ScheduledRootHash:     hex.EncodeToString(epochStartShardData.ScheduledRootHash),
+		FirstPendingMetaBlock: hex.EncodeToString(epochStartShardData.FirstPendingMetaBlock),
+		LastFinishedMetaBlock: hex.EncodeToString(epochStartShardData.LastFinishedMetaBlock),
+	}
+
+	if len(epochStartShardData.PendingMiniBlockHeaders) == 0 {
+		apiBlock.EpochStartShardsData = append(apiBlock.EpochStartShardsData, shardData)
```

Comment:
move this outside the if branch and remove L222

Reply:
done.